redfish_client/resource.py

```diff
@@ -149,3 +156,32 @@ def wait_for(self, stat, expected, blacklisted=None, poll_interval=3, timeout=15
     @property
     def raw(self):
         return self._content
+
+    def post(self, attribute=None, payload=None):
+        """
+        Perform a POST at the resource with the given payload.
+
+        Args:
+          attribute: Optional attribute to POST to
+          payload: The contents of the POST payload.
+        """
+        if attribute is None:
+            path = self._content.get("@odata.id")
+        else:
+            path = self._content[attribute].get("@odata.id")
+        if path:
+            return self._connector.post(path, payload=payload)
+        else:
```

Comment:
This is redundant, since we return from the `True` branch.

Reply:
Technically, I agree, but I prefer this for legibility.

Reply:
We could also do

```python
if not path:
  raise MissingOidException("The resource cannot be POSTed to.")
return self._connector.post(path, payload=payload)
```

Reply:
EDIT: I see we're repeating this iod parsing many times. May I propose we introduce auxilary function for this purpose?

```python
@property
def oid(self):
  oid = self.raw.get("@odata.id")
  if not oid:
    raise MissingOidException("The resource has no oid.")
  return oid
end

def post(self, payload=None):
  return self._connector.post(self.oid, payload=payload)
end

def patch(self, payload):
  return self._connector.patch(self.oid, payload=payload)
end

...
```

Reply:
👍

Reply:
Good idea, but I'd like to move on rather than continue to polish the commits too much. Perhaps worth doing in the upcoming changes.